lib/src/puppetdb-cli.cc

```diff
@@ -50,8 +56,20 @@ parse_config() {
 
 string
 pdb_server_url(const json::JsonContainer& config) {
-    const auto server_urls = config.get< vector<string> >("server_urls");
-    return server_urls.size() ? server_urls[0] : "http://127.0.0.1:8080";
+    const auto server_urls_datatype = config.type("server_urls");
+    if (server_urls_datatype == json::DataType::Array) {
+        const auto server_urls = config.get< vector<string> >("server_urls");
+        return server_urls.size() ? server_urls[0] : "http://127.0.0.1:8080";
+    } else if (server_urls_datatype == json::DataType::String){
+        return config.get< string >("server_urls");
+    } else {
+        logging::colorize(nowide::cerr, logging::log_level::fatal);
+        nowide::cerr << "error: unrecognized JSON type for key 'server_urls'." << "\n"
+                     << "Please use a JSON array or string."<< "\n"
+                     << "falling back to default configuration" << "\n"<< endl;
+        logging::colorize(nowide::cerr);
+        return "http://127.0.0.1:8080";
```

Comment:
My preference here would be a hard fail. If the user has given us a bad server_url, we need to get that straightened out. This seems like it would give the nice error message followed by something that would probably be bad (trying to connect to a PuppetDB that doesn't exist). This also removes a third place where we're provide a default.